closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1070,23 +887,615 @@ boolean isRestrictedProviderAllowed(String providerName) {
+            System.out.println();
+        }
+
+        private static void printProperty(String name, String value) {
+            if (value != null) {
+                String valueToPrint = (value.isEmpty()) ? "EMPTY" : value;
+                System.out.println(name + valueToPrint);
+            } else if (debug != null) {
+                debug.println("Nothing to print. Value of property " + name + " is null.");
+            }
+        }
+    }
+
+    private static final class ProfileParser {
+        // Properties specified through the profile.
+        private final Map<String, String> profileProperties;
+        private boolean descIsDefault;
+        private boolean descIsFIPS;
+
+        // Provider with argument (provider name + optional argument).
+        private final List<String> providers;
+        // Provider without argument.
+        private final List<String> providersSimpleName;
+        // The map is keyed by provider name.
+        private final Map<String, List<Constraint>> providerConstraints;
+
+        private final String profileID;
+
+        private Map<String, String> profilesHashes;
+        private Map<String, List<String>> profilesInfo;
+
+        // The java.security properties.
+        private final Properties securityProps;
+
+        private final List<String> appendableProps = Arrays.asList("jdkTlsDisabledNamedCurves",
+                                                                   "jdkTlsDisabledAlgorithms",
+                                                                   "jdkTlsLegacyAlgorithms",
+                                                                   "jdkCertpathDisabledAlgorithms",
+                                                                   "jdkSecurityLegacyAlgorithms");
+
+        /**
          *
-         * @param providerName provider name or provider with packages
-         * @return provider name defined in provider construction method
+         * @param id    the restricted security custom profile ID
+         * @param props the java.security properties
+         */
+        private ProfileParser(String id, Properties props) {
+            Objects.requireNonNull(props);
+
+            profileID = id;
+            securityProps = props;
+
+            profileProperties = new HashMap<>();
+
+            providers = new ArrayList<>();
+            providersSimpleName = new ArrayList<>();
+            providerConstraints = new HashMap<>();
+
+            profilesHashes = new HashMap<>();
+            profilesInfo = new HashMap<>();
+
+            // Initialize the properties.
+            init(profileID);
+        }
+
+        private RestrictedSecurityProperties getProperties() {
+            RestrictedSecurityProperties rsProperties = new RestrictedSecurityProperties();
+
+            rsProperties.profileID = profileID;
+
+            rsProperties.descName = getProperty("descName");
```

Comment:
This should be done within the constructor for `RestrictedSecurityProperties` rather than from the outside (it will also allow fields like `profileID` to be `final`.

Reply:
Moved most fields to the constructors. I'm not sure about the lists and map though. Think I should move those too?